docs/azure/sdk/unit-testing-mocking.md

```diff
@@ -231,6 +231,36 @@ var finder = new AboutToExpireSecretFinder(TimeSpan.FromDays(2), secretClient);
 
 This approach is useful when you would like to consolidate the dependency creation and share the client between multiple consuming classes.
 
+## Understand Azure Resource Manager clients
+
+In Azure Resource Manager libraries, the clients were designed to emphasize their relationship to one another, mirroring the service hierarchy. To achieve that goal, extension methods are widely used to add additional features to clients.
+
+For example, an Azure virtual machine resizes in an Azure resource group. The `Azure.ResourceManager.Compute` namespace models Azure virtual machine as `VirtualMachineResource` and `Azure.ResourceManager` namespace models Azure resource group as `ResourceGroupResource`.  To query the virtual machines for a resource group, you would write:
+
+:::code language="csharp" source="snippets/unit-testing/ResourceManager/ResourceManagerCodeStructure.cs" id="ParentOfVMIsRG" :::
+
+Because the virtual machine-related functionality such as `GetVirtualMachines` on `ResourceGroupResource` are implemented as extension methods, it is not possible to just create a mock of the type and override the method. Instead, you'll also have to create a mock class for the "mockable resource" and wire them together.
```

Comment:
```suggestion
Because the virtual machine-related functionality such as `GetVirtualMachines` on `ResourceGroupResource` are implemented as extension methods, it's not possible to just create a mock of the type and override the method. Instead, you'll also have to create a mock class for the "mockable resource" and wire them together.
```